ios/gamma/split-view/RNSSplitViewScreenController.swift

```diff
@@ -69,46 +87,144 @@ public class RNSSplitViewScreenController: UIViewController {
   /// to the Host view from the modal case where we're passing absolute layout metrics to the ShadowNode.
   ///
-  @objc
-  public override func viewDidLayoutSubviews() {
-    super.viewDidLayoutSubviews()
+  /// Prefers to apply dynamic updates from the presentation layer if the transition is in progress.
+  ///
+  private func updateShadowTreeState() {
     // For modals, which are presented outside the SplitViewHost subtree (and RN hierarchy),
-    // we're attaching our touch handler adn we don't need to apply any offset corrections,
+    // we're attaching our touch handler and we don't need to apply any offset corrections,
     // because it's positioned relatively to our RNSSplitViewScreenComponentView
-    if isInSplitViewHostSubtree() {
-      let ancestorView = findSplitViewHostController()?.view
+    if !isInSplitViewHostSubtree() {
+      shadowStateProxy.updateShadowState(ofComponent: splitViewScreenComponentView)
+      return
+    }
 
-      assert(
-        ancestorView != nil,
-        "[RNScreens] Expected to find RNSSplitViewHost component for RNSSplitViewScreen component"
+    let ancestorView = findSplitViewHostController()?.view
+    assert(
+      ancestorView != nil,
+      "[RNScreens] Expected to find RNSSplitViewHost component for RNSSplitViewScreen component"
+    )
+
+    // If the resize animation is currently running, we prefer to apply dynamic updates,
+    // based on the results from the presentation layer
+    // which is read from `trackTransitionProgress` method.
+    if let currentSize = lastAnimationFrame?.size {
+      applyTransitioningShadowState(
+        size: currentSize,
+        ancestorView: ancestorView!
       )
+      return
+    }
 
-      shadowStateProxy.updateShadowState(
-        ofComponent: splitViewScreenComponentView,
-        inContextOfAncestorView: ancestorView!
-      )
-    } else {
-      shadowStateProxy.updateShadowState(
-        ofComponent: splitViewScreenComponentView
-      )
+    // There might be the case, when transition is about to start and in the meantime,
+    // sth else is triggering frame update relatively to the parent. As we know
+    // that dynamic updates from the presentation layer are coming, we're blocking this
+    // to prevent interrupting with the frames that are less important for us.
+    // This works fine, because after the animation completion, we're sending the last update
+    // which is compatible with the frame which would be calculated relatively to the ancestor here.
+    if !isTransitionInProgress() {
+      applyStaticShadowStateRelativeToAncestor(ancestorView!)
     }
   }
 
+  private func applyTransitioningShadowState(size: CGSize, ancestorView: UIView) {
+    let localOrigin = splitViewScreenComponentView.convert(
+      splitViewScreenComponentView.frame.origin,
+      to: ancestorView
+    )
+    let convertedFrame = CGRect(origin: localOrigin, size: size)
+    shadowStateProxy.updateShadowState(withFrame: convertedFrame)
```

Comment:
why not use `- [RNSSplitViewScreenShadowStateProxy updateShadowState:inContextOfAncestorView:]`? Why reimplement the method here?